dojo/tools/bugcrowd_api/parser.py

```diff
@@ -0,0 +1,275 @@
+                char_to_replace = {":": " ", '"': " ", "@": "at"}
+                for key, value in char_to_replace.items():
+                    title = title.replace(key, value)
+
+            date = dateutil.parser.parse(entry["attributes"]["submitted_at"])
+
+            bug_url = ""
+            if entry["attributes"]["bug_url"]:
+                try:
+                    if "://" in entry["attributes"]["bug_url"]:  # is the host full uri?
+                        bug_endpoint = Endpoint.from_uri(
+                            entry["attributes"]["bug_url"].strip()
+                        )
+                        # can raise exception if the host is not valid URL
+                    else:
+                        bug_endpoint = Endpoint.from_uri(
+                            "//" + entry["attributes"]["bug_url"].strip()
+                        )
+                        # can raise exception if there is no way to parse the host
+                except Exception as e:  # We don't want to fail the whole import just for 1 error in the bug_url
+                    logger.error(
+                        "Error parsing bugrcrowd bug_url : {}".format(
+                            entry["attributes"]["bug_url"].strip()
+                        )
+                    )
+                bug_url = entry["attributes"]["bug_url"]
+
+            description = "\n".join(
+                [
+                    entry["attributes"]["description"],
+                    "",
+                    "Bugcrowd details:",
+                    f"- Severity: P{ bugcrowd_severity }",
+                    f"- Bug Url: { bug_url }",
+                    "",
+                    f"Bugcrowd link: {links}",
+                ]
+            )
+            mitigation = entry["attributes"]["remediation_advice"]
+            steps_to_reproduce = entry["attributes"]["description"]
+            unique_id_from_tool = entry["id"]
+
+            finding = Finding(
+                test=test,
+                title=textwrap.shorten(title, width=511, placeholder="..."),
+                date=date,
+                severity=self.convert_severity(bugcrowd_severity),
+                description=description,
+                mitigation=mitigation,
+                steps_to_reproduce=steps_to_reproduce,
+                active=self.is_active(bugcrowd_state),
+                verified=self.is_verified(bugcrowd_state),
+                false_p=self.is_false_p(bugcrowd_state),
+                duplicate=bugcrowd_duplicate,
+                out_of_scope=self.is_out_of_scope(bugcrowd_state),
+                risk_accepted=self.is_risk_accepted(bugcrowd_state),
+                is_mitigated=self.is_mitigated(bugcrowd_state),
+                static_finding=False,
+                dynamic_finding=True,
+                unique_id_from_tool=unique_id_from_tool,
+            )
+
+            if bug_endpoint:
+                try:
+                    finding.unsaved_endpoints = [bug_endpoint]
```

Comment:
This also has a little bit of misleading error message. Because if there will be an `Exception` it will be because of creating list and/or setting variable. There will be no problem with _the parsing_ of an endpoint.